.github/workflows/main.yml

```diff
@@ -3,10 +3,25 @@ on:
   push:
     branches:
       - main
+  pull_request:
+    branches:
+      - main
     tags:
       - '*.*.*'
 
 jobs:
+  test:
+    name: Run unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - 
+        name: Check out the repo
+        uses: actions/checkout@v4
+      - 
+        name: Unit testing 
+        uses: fylein/python-pytest-github-action@v2
+        with:
+          args: pip3 install -r requirements.txt && pytest
```

Comment:
you should probably add a `    needs: test` to your build action as it should not run if the tests fail.

Reply:
at some point it was there, will add it back...